<launch>

  <test time-limit="10" test-name="footprint_test" pkg="costmap_2d" type="footprint_tests">
    <param name="unpadded/string/footprint_padding" value="0" />
    <param name="unpadded/footprint" value="[[1, 1], [-1, 1], [-1, -1]]" />

    <param name="padded/string/footprint_padding" value=".5" />
    <param name="padded/footprint" value="[[1, 1], [-1, 1], [-1, -1]]" />

    <param name="radius/sub/footprint_padding" value="0" />
    <param name="radius/robot_radius" value="10" />

    <rosparam ns="xmlrpc">
      footprint_padding: 0
      footprint: [[0.1, 0.1], [-0.1, 0.1], [-0.1, -0.1], [0.1, -0.1]]
    </rosparam>

    <rosparam ns="xmlrpc_fail"> <!-- Footprint includes a 3-value point, which should make it fail. -->
      footprint_padding: 0
      footprint: [[0.1, 0.1], [-0.1, 0.1, 77.0], [-0.1, -0.1], [0.1, -0.1]]
    </rosparam>

    <param name="same_level/footprint_padding" value="0" />
    <param name="same_level/footprint" value="[[1, 2], [3, 4], [5, 6]]" />

    <param name="empty/empty" value="0" /> <!-- just so you can see there are no real params under "empty". -->
  </test>

</launch>
